source/steamshim/src/os.cpp

```diff
@@ -60,17 +71,25 @@ bool setEnvVar(const char *key, const char *val)
 
 bool launchChild(ProcessType *pid, const char* name)
 {
-    char *str = _strdup( GetCommandLineA() );
-
     STARTUPINFOA si = { sizeof( si ) };
     memset( pid, 0, sizeof( *pid ) );
 
     char exename[32] = ".\\";
     strncpy(exename+2, name, 30);
 
-    bool bResult = ( CreateProcessA( exename, str, NULL, NULL, TRUE, 0, NULL,
+    char args[32] = { 0 };
+    if (debug){
+        snprintf(args, sizeof(args) - 1, ".\\%s steamdebug", name);
+    } else {
+        snprintf(args, sizeof(args) - 1, ".\\%s", name);
+    }
+    
+    DWORD dwAttrib = GetFileAttributesA(exename);
+    if (dwAttrib == INVALID_FILE_ATTRIBUTES ||
+         (dwAttrib & FILE_ATTRIBUTE_DIRECTORY)) return false;
+
+    bool bResult = ( CreateProcessA( exename, args, NULL, NULL, TRUE, 0, NULL,
```

Comment:
Because `args` already includes the executable name, you can change the first argument to NULL.